tests/test_json.ml

```diff
@@ -0,0 +1,60 @@
+let read_file fname =
+  (* the tests are run from the _build/default/tests directory *)
+  Fpath.(v ".." / ".." / ".." / "tests" / fname)
+  |> Fpath.to_string |> Yojson.Basic.from_file
+
+let dune_release_testing_json = read_file "dune-release-testing.json"
+
+let upload_response_json = read_file "upload-response.json"
+
+let create_release_response_json = read_file "create-release-response.json"
+
+let pull_request_response_json = read_file "pull-request-response.json"
+
+let archive_upload_url () =
+  let check json expected =
+    match Dune_release.Json.Upload_response.browser_download_url json with
+    | Ok actual -> Alcotest.(check string) __LOC__ actual expected
+    | Error (`Msg msg) -> Alcotest.fail msg
+  in
+  check dune_release_testing_json
+    "https://github.com/NathanReb/dune-release-testing/releases/download/v0.0.0/dummy-v0.0.0.tbz";
+  check upload_response_json
+    "https://github.com/octocat/Hello-World/releases/download/v1.0.0/example.zip"
+
+let release_id () =
+  let check json expected =
+    match Dune_release.Json.Release_response.release_id json with
+    | Ok actual -> Alcotest.(check int) __LOC__ actual expected
```

Comment:
I would expect `Alcotest` to have an `'a testable -> 'a result testable` combinator. If it doesn't I think we should add a `Testable` module in our tests to add our own custom `testable`s. I've been doing this for a few projects and that works fairly well as it makes it a natural place to add `testable`s for our own types as well!

Reply:
Done